perception/sub8_pointcloud/src/OGridGen.cpp

```diff
@@ -52,6 +54,37 @@ void OGridGen::publish_ogrid(const ros::TimerEvent &)
     cv::polylines(mat_ogrid_, &pts, &npts, 1, true, 255, 3, CV_8U, 0);
   }
 
+
+  // Populate a PCL pointcloud using the point_cloud_buffer_
+  pcl::PointCloud<pcl::PointXYZI>::Ptr pointCloud(new pcl::PointCloud<pcl::PointXYZI>());
+  pointCloud->reserve(point_cloud_buffer_.capacity());
+  for (auto &p : point_cloud_buffer_)
+  {
+    pointCloud->push_back(p);
+  }
+  
+  // Publish the point cloud
+  pointCloud->header.frame_id = "map";
+  pcl_conversions::toPCL(ros::Time::now(), pointCloud->header.stamp);
+  pub_point_cloud_raw_.publish(pointCloud);
+
+  pcl::PointCloud<pcl::PointXYZI>::Ptr pointCloud_filtered = classification_.filtered(pointCloud);
+  pub_point_cloud_filtered_.publish(pointCloud_filtered);
+
+  // Do some algorithms using the PCL pointcloud
+  // classification_.segment(pointCloud);
+  // classification_.findNormals(pointCloud);
+
+  // Populate the Ogrid by projecting down the pointcloud
+  for(auto &point_pcl : pointCloud_filtered->points) {
+    cv::Rect rect(cv::Point(), mat_ogrid_.size());
```

Comment:
Are you sure Point() will always be initialized to (0, 0), I think it's best to be explicit here.